rpc/block_svc_routines.c

```diff
@@ -4343,6 +4349,45 @@ block_create_common(blockCreate *blk, char *rbsize, char *volServer, char *prio_
                     GB_TGCLI_GLFS_PATH, blk->block_name) == -1) {
       goto out;
     }
+
+    /*
+     * The ALUA states and transition:
+     *
+     * a) active/optimized: supported
+     * b) active/non-optimized: supported
+     * c) standby: non-supported
+     * d) unavailable: non-supported
+     * e) offline: non-supported
+     *
+     * f) transition: supported
+     */
+    if (GB_ASPRINTF(&glfs_alua_sup,
+                    "%s/%s/alua/glfs_tg_pt_gp_ao set alua alua_support_active_optimized=1\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ao set alua alua_support_active_nonoptimized=1\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ao set alua alua_support_offline=0\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ao set alua alua_support_standby=0\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ao set alua alua_support_unavailable=0\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ao set alua alua_support_transitioning=1"
+                    "%s/%s/alua/glfs_tg_pt_gp_ano set alua alua_support_active_optimized=1\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ano set alua alua_support_active_nonoptimized=1\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ano set alua alua_support_offline=0\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ano set alua alua_support_standby=0\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ano set alua alua_support_unavailable=0\n"
+                    "%s/%s/alua/glfs_tg_pt_gp_ano set alua alua_support_transitioning=1",
+                    GB_TGCLI_GLFS_PATH, blk->block_name,
```

Comment:
I think by default everything is enabled.
So we don't have to explicitly enable alua_support_active_optimized, alua_support_active_nonoptimized and alua_support_transitioning states, this reduces 6 less settings and hence the time to create per block volume will be less impacted.

Reply:
Not sure, this should be depended with different kernel version.

And there is no need to build the cache for each of them. So for the time shouldn't be much impacted IMO.

Thanks.

Reply:
@lxbsz 

Can we have a check on this ? IMHO if weare not changing anything currently about this in the kernel, and atleast since quite sometime now, we should be okay to avoid extra settings.

Also you need to apply the same to genconfig logic as well.

Reply:
Okay.
Will check and remove it.
Thanks.